src/librustdoc/clean/types.rs

```diff
@@ -73,8 +75,8 @@ crate struct ExternalCrate {
     crate name: Symbol,
     crate src: FileName,
     crate attrs: Attributes,
-    crate primitives: Vec<(DefId, PrimitiveType)>,
-    crate keywords: Vec<(DefId, Symbol)>,
+    crate primitives: ThinVec<(DefId, PrimitiveType)>,
+    crate keywords: ThinVec<(DefId, Symbol)>,
```

Comment:
Why not make it thinner by combining this two into one pointer?

Reply:
What do you mean? These are stored in the ThinVec, on the heap, not on the stack.

Reply:
Yes, but since both of them are as rare, so we can just combine both into one pointer rather than two right? But yeah, if one is needed then we need the pointer already.

Reply:
It's possible that combining them into one boxed field would make it smaller, but

1. that would require code changes, and the slight perf improvement is likely not worth it;
2. there probably won't be much of a perf improvement because padding may take up the space anyway.